.github/workflows/test.yml

```
@@ -1,5 +1,11 @@
 name: test
-on: [push]
+on:
+  pull_request:
+    branches: [main]
```

I'd still keep this part

```suggestion
  push:
    branches:
      - main
  pull_request: {}
```

---

I removed it because pull request checks run in the context of the merge commit, so running on push to main is redundant. Is there another use case you have in mind?

---

I always thought that the push to main builds and the pull requests builds were slightly different but maybe this is no longer the case.

The other reason is if we want to add a badge showing that the CI passes, not sure if we could show a badge if we don't run test builds on the main branch?